### ISSUE TITLE
ValueError: invalid literal for int() with base 10: ''

### ISSUE THREAD

**Issue by thuer-it** [jamesridgway/attachment-downloader#14]
Hi,

One additional error, I am getting, but not sure why.

The email B29 did not have an attachment, but others also do not have one
` 2021-01-28 22:38:51,879 - INFO - Processing message 'b'29'' subject 'Würden Sie uns auch bewerten? Semaf Electronics'
Traceback (most recent call last):
  File "/home/andreas/.local/bin/attachment-downloader", line 69, in <module>
    for (uid, message) in messages:
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/messages.py", line 55, in _fetch_email_list
    yield uid, self._fetch_email(uid)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/messages.py", line 42, in _fetch_email
    return fetch_email_by_uid(uid=uid,
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 155, in fetch_email_by_uid
    email_object = parse_email(raw_email, policy=parser_policy)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 212, in parse_email
    attachment = parse_attachment(part)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 122, in parse_attachment
    filename_parts.insert(int(s_name[1]),value[1:-1] if value.startswith('"') else value)
ValueError: invalid literal for int() with base 10: ''`

If I am deleting the “b29”, I am getting the same error on an other email:
`
2021-01-28 22:50:41,183 - INFO - Downloaded and parsed mail 'Fwd: Ihre Rechnung von ABC IN2021993' with 1 attachments
2021-01-28 22:50:41,183 - INFO - Mark UID 26 with \Seen FLAG
2021-01-28 22:50:41,195 - INFO - Processing message 'b'26'' subject 'Fwd: Ihre Rechnung von ABC IN2021993'
2021-01-28 22:50:41,196 - INFO - Downloading attachment 'IN2021993.pdf' to path /home/andreas/Downloads/mail/IN2021993.pdf
2021-01-28 22:50:41,196 - WARNING - Overwriting file: '/home/andreas/Downloads/mail/IN2021993.pdf'
Traceback (most recent call last):
  File "/home/andreas/.local/bin/attachment-downloader", line 69, in <module>
    for (uid, message) in messages:
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/messages.py", line 55, in _fetch_email_list
    yield uid, self._fetch_email(uid)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/messages.py", line 42, in _fetch_email
    return fetch_email_by_uid(uid=uid,
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 155, in fetch_email_by_uid
    email_object = parse_email(raw_email, policy=parser_policy)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 212, in parse_email
    attachment = parse_attachment(part)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 122, in parse_attachment
    filename_parts.insert(int(s_name[1]),value[1:-1] if value.startswith('"') else value)
ValueError: invalid literal for int() with base 10: '' `

Br
Andreas

**Comment by jamesridgway** [comment]
Can you check if this is still an issue with 1.1.3 which has just been released?

Also, how many emails exist in the folder that you're searching through? I've struggled to recreated the issue so far.

**Comment by arillus-at** [comment]
Hi,

I tried to reproduce with V1.1.3. The issue changes now to this messages:

**Mail without attachment:**
```
2021-02-16 12:54:46,909 - INFO - Downloaded and parsed mail 'Domaintechnik.at Erinnerung: Dokumente erforderlich' with 0 attachments
Traceback (most recent call last):
  File "/home/andreas/.local/bin/attachment-downloader", line 116, in <module>
    parsed_message_date = datetime.datetime.strptime(sanitised_date, '%a, %d %b %Y %H:%M:%S %z')
  File "/usr/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/lib/python3.8/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '15 Feb 2021 06:30:39 -0000' does not match format '%a, %d %b %Y %H:%M:%S %z'
```


**Mail with "()" in subject:**
```
Traceback (most recent call last):
  File "/home/andreas/.local/bin/attachment-downloader", line 114, in <module>
    for (uid, message) in messages:
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/messages.py", line 55, in _fetch_email_list
    yield uid, self._fetch_email(uid)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/messages.py", line 42, in _fetch_email
    return fetch_email_by_uid(uid=uid,
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 155, in fetch_email_by_uid
    email_object = parse_email(raw_email, policy=parser_policy)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 212, in parse_email
    attachment = parse_attachment(part)
  File "/home/andreas/.local/lib/python3.8/site-packages/imbox/parser.py", line 122, in parse_attachment
    filename_parts.insert(int(s_name[1]),value[1:-1] if value.startswith('"') else value)
ValueError: invalid literal for int() with base 10: ''
```

**Comment by jensstruemper** [comment]
Hi! 
I still get the same issue with version 1.1.6

2021-05-06 21:54:33,256 - INFO - Processing message 'b'47'' subject '✅ Kauf bestätigt: ...'
Traceback (most recent call last):
  File "/usr/local/bin/attachment-downloader", line 140, in <module>
    for (uid, message) in messages:
  File "/usr/local/lib/python3.9/site-packages/imbox/messages.py", line 55, in _fetch_email_list
    yield uid, self._fetch_email(uid)
  File "/usr/local/lib/python3.9/site-packages/imbox/messages.py", line 42, in _fetch_email
    return fetch_email_by_uid(uid=uid,
  File "/usr/local/lib/python3.9/site-packages/imbox/parser.py", line 155, in fetch_email_by_uid
    email_object = parse_email(raw_email, policy=parser_policy)
  File "/usr/local/lib/python3.9/site-packages/imbox/parser.py", line 212, in parse_email
    attachment = parse_attachment(part)
  File "/usr/local/lib/python3.9/site-packages/imbox/parser.py", line 122, in parse_attachment
    filename_parts.insert(int(s_name[1]),value[1:-1] if value.startswith('"') else value)
ValueError: invalid literal for int() with base 10: ''

**Comment by jamesridgway** [comment]
I've tested this using unicode characters in the email and in the filename of attachments and I cannot reproduce the issue.

Can I ask if you still get the issue with 1.2.2?

**Comment by jamesridgway** [comment]
Closing due to no reply

**Comment by RodrigoVanzelotti** [comment]
Hey @jamesridgway, I'm into the same the same problem. @jensstruemper have you found a proper solution?
The only thing is that I don't have a "Mail with "()" in subject:" as Jenss sugested.
- My imbox version is 0.9.8
When I try to update it with `pip install imbox --upgrade` or `pip install imbox -U` it just returns:
```
Requirement already satisfied: imbox in c:\users\rodri\appdata\local\programs\python\python310\lib\site-packages (0.9.8)
Requirement already satisfied: chardet in c:\users\rodri\appdata\local\programs\python\python310\lib\site-packages (from imbox) (3.0.4)

[notice] A new release of pip is available: 23.1.2 -> 23.2.1    
[notice] To update, run: python.exe -m pip install --upgrade pip
```

Here's my code:
A little context. It gets the attachments and sort them in folders in a bigger folder called `attachments`. It does the sorting by domain name. The `self.filter_name` filters chars that might not be accepted by the OS when saving a file, such as `'[<>:"/\\|?*]'`.
```python
with Imbox(self.host, username=self.email, password=self.password) as imbox:
            messages = imbox.messages(folder="inbox", date__gt=date_since,  raw='has:attachment')
        
            for uid, message in messages:
                # imbox.mark_seen(uid)
                if message.attachments:
                    sender_email = message.sent_from[0]["email"]

                    # Extract domain from sender's email and check if domain name is ok for creating folder
                    domain = sender_email.split('@')[1].split('.')[0]
                    domain = self.filter_name(domain)

                    # Create subdirectory based on domain if it doesn't exist
                    domain_dir = os.path.join(self.attachments, domain)
                    if not os.path.exists(domain_dir):
                        os.mkdir(domain_dir)

                    for attachment in message.attachments:
                        # Save attachments to the designated domain subdirectory
                        file_data = attachment.get('content')
                        filename = self.filter_name(attachment.get('filename'))

                        if filename == '':
                            filename = dateprint + str(error_counter) + '.pdf'
                            error_counter += 1

                        filepath = os.path.join(domain_dir, filename)

                        with open(filepath, "wb") as f:
                            f.write(file_data.read())

                        if self.con_lvl >= 3:
                            print(f"Saved attachment {filename} for message {uid} from {sender_email}")
                            print(f'\t\tPATH: {filepath}')

                        session_counter += 1
```
And the error:
```python
Traceback (most recent call last):
  File "c:\Users\rodri\OneDrive\Área de Trabalho\Projetos\Asimov\COPE\email_consumer.py", line 104, in <module>
    Mailer.get_attachments(date_since)
  File "c:\Users\rodri\OneDrive\Área de Trabalho\Projetos\Asimov\COPE\email_consumer.py", line 39, in get_attachments
    for uid, message in messages:
  File "C:\Users\rodri\AppData\Local\Programs\Python\Python310\lib\site-packages\imbox\messages.py", line 55, in _fetch_email_list
    yield uid, self._fetch_email(uid)
  File "C:\Users\rodri\AppData\Local\Programs\Python\Python310\lib\site-packages\imbox\messages.py", line 42, in _fetch_email
    return fetch_email_by_uid(uid=uid,
  File "C:\Users\rodri\AppData\Local\Programs\Python\Python310\lib\site-packages\imbox\parser.py", line 155, in fetch_email_by_uid
    email_object = parse_email(raw_email, policy=parser_policy)
  File "C:\Users\rodri\AppData\Local\Programs\Python\Python310\lib\site-packages\imbox\parser.py", line 212, in parse_email
    attachment = parse_attachment(part)
  File "C:\Users\rodri\AppData\Local\Programs\Python\Python310\lib\site-packages\imbox\parser.py", line 122, in parse_attachment
    filename_parts.insert(int(s_name[1]),value[1:-1] if value.startswith('"') else value)
ValueError: invalid literal for int() with base 10: ''
```
The program reads some of the attachments but struggles in this part. The error seem to be in the unpacking of the next `message` in the iteration process. 

Thanks for your time,
Vanzelotti
